src/renderer/services/file-storage-service/index.ts

```diff
@@ -114,25 +114,68 @@ export default class FileStorageService extends HttpCacheClient {
    * This is an incremental upload act, after an upload has been registered
    * the file can be send in chunked of pretermined size to the service.
    */
-  public sendUploadChunk(
+  public async sendUploadChunk(
     uploadId: string,
     chunkNumber: number,
     rangeStart: number,
     md5ThusFar: string,
     postBody: Uint8Array,
     user: string
-  ): Promise<UploadChunkResponse> {
+  ): Promise<void> {
     const url = `${FileStorageService.BASE_UPLOAD_PATH}/${uploadId}/chunk/${chunkNumber}`;
     const rangeEnd = rangeStart + postBody.byteLength - 1;
-    return this.post<UploadChunkResponse>(url, postBody, {
-      ...FileStorageService.getHttpRequestConfig(),
-      headers: {
-        "Content-Type": "application/octet-stream",
-        "Cumulative-MD5": md5ThusFar,
-        Range: `bytes=${rangeStart}-${rangeEnd}`,
-        "X-User-Id": user,
-      },
-    });
+    const attemptRequest = () => (
+      this.post<UploadChunkResponse>(url, postBody, {
+        ...FileStorageService.getHttpRequestConfig(),
+        headers: {
+          "Content-Type": "application/octet-stream",
+          "Cumulative-MD5": md5ThusFar,
+          Range: `bytes=${rangeStart}-${rangeEnd}`,
+          "X-User-Id": user,
+        },
+      }
+      )
+    );
+    try {
+      await attemptRequest();
+    } catch (error) {
+      // Re-throw error if status can't be determine (like when the error doesn't have to do with the server/HTTP request
+      // or if the status is between acceptable ranges indicating another problem is afoot
+      if (!error.response || !error.response.status || (error.response.status >= 200 && error.response.status < 300)) {
+        throw error;
+      }
+
+      // Currently there is a bug in FSS that causes the upload chunk endpoint
```

Comment:
> Currently there is a bug in FSS that causes the upload chunk endpoint

Currently there _are infrastructure performance bottlenecks_ that causes the upload chunk endpoint 

?